### ISSUE TITLE
[BUG]: Error when using Prisma with Shopify

### ISSUE THREAD

**Issue by terciodejesus** [remix-run/remix#5911]
### What version of Remix are you using?

2023.1.2

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

- Use Shopify with Remix.
- Install and build prisma.
- Use PrismaClient in app/utils/db.server.ts
- npm run dev

```
//app/utils/db.server.ts
//@ts-nocheck
import {PrismaClient} from '@prisma/client';

let prisma: PrismaClient;
declare global {
  var __db: PrismaClient | undefined;
}

if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient();
  prisma.$connect();
} else {
  if (!global.__db) {
    global.__db = new PrismaClient();
    global.__db.$connect();
  }
  prisma = global.__db;
}

export {prisma};

```

### Expected Behavior

new PrismaClient() should run without problems.

### Actual Behavior

```
$ npm run dev

> demo-store@0.0.0 dev
> npm run build:css && concurrently -g --kill-others-on-fail -r npm:dev:css "npm run sass" "run-s icons3 & npm-watch icons"  "shopify hydrogen dev"


> demo-store@0.0.0 build:css
> postcss styles --base styles --dir app/styles --env production

/bin/sh: 1: run-s: not found
[icons] [nodemon] 2.0.20

> demo-store@0.0.0 sass
> sass --watch --no-source-map  app/:app/


> demo-store@0.0.0 dev:css
> postcss styles --base styles --dir app/styles -w

Sass is watching for changes. Press Ctrl-C to stop.

Initial build: 1.552s
🚥 MiniOxygen server started at http://localhost:3000


╭─ info ───────────────────────────────────────────────────────────────────────╮
│                                                                              │
│  Upgrade available                                                           │
│                                                                              │
│  Version 2023.1.6 of @shopify/hydrogen is now available.                     │
│                                                                              │
│  You are currently running v2023.1.2.                                        │
│                                                                              │
│  Reference                                                                   │
│    • Hydrogen releases                                                       │
│                                                                              │
╰──────────────────────────────────────────────────────────────────────────────╯

[icons] [nodemon] clean exit - waiting for changes before restart
/home/tercio/dev/ecommerce-hydrogen/node_modules/yoga-layout-prebuilt/yoga-layout/build/Release/nbind.js:53
        throw ex;
        ^

Error: PrismaClient is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
    at new PrismaClient2 (/home/tercio/dev/ecommerce-hydrogen/node_modules/.prisma/client/index-browser.js:189:11)
    at /home/tercio/dev/ecommerce-hydrogen/app/utils/db.server.ts:14:19
    at SourceTextModule.evaluate (node:internal/vm/module:226:23)
    at VMScriptRunner.runAsModule (/home/tercio/dev/ecommerce-hydrogen/node_modules/@miniflare/runner-vm/src/index.ts:38:18)
    at VMScriptRunner.run (/home/tercio/dev/ecommerce-hydrogen/node_modules/@miniflare/runner-vm/src/index.ts:82:17)
    at MiniOxygen.#reload (/home/tercio/dev/ecommerce-hydrogen/node_modules/@miniflare/core/src/index.ts:775:13)

Node.js v18.12.1
```

**Comment by j-acks0n** [comment]
Hi! Have you been able to figure out why this happens?

**Comment by Hector101** [comment]
This seems to be an intentional implementation from Shopify hydrogen cli, cos it works fine on a regular remix app.

**Comment by terciodejesus** [comment]
> This seems to be an intentional implementation from Shopify hydrogen cli, cos it works fine on a regular remix app.

@Hector101  Yes. I have tried to use Prisma with Remix app and it works fine. But I didn't have success with the Remix Shopify app

**Comment by jlemrond** [comment]
I am seeing the same issue with `@shopify/hydrogen`.  

Also, working on another Remix app that does not use hydrogen where Prisma is working perfectly. That app is being deployed to Netlify.

```
🏁 Initial build: 719.778ms
🚥 MiniOxygen server started at http://localhost:3000 ( http://localhost:3000 )

Error: PrismaClient is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
    at new PrismaClient2 (/Users/jasonlemrond/Documents/projects/tearribles/web/node_modules/.pnpm/@prisma+client@4.13.0-dev.41_prisma@4.13.0-dev.41/node_modules/.prisma/client/index-browser.js:118:11)
    at /Users/jasonlemrond/Documents/projects/tearribles/web/app/utils/db.server.ts:3:16
    at SourceTextModule.evaluate (node:internal/vm/module:226:23)
    at VMScriptRunner.runAsModule (/Users/jasonlemrond/Documents/projects/tearribles/web/node_modules/.pnpm/@miniflare+runner-vm@2.13.0/node_modules/@miniflare/runner-vm/src/index.ts:40:18)
    at VMScriptRunner.run (/Users/jasonlemrond/Documents/projects/tearribles/web/node_modules/.pnpm/@miniflare+runner-vm@2.13.0/node_modules/@miniflare/runner-vm/src/index.ts:86:17)
    at MiniOxygen.#reload (/Users/jasonlemrond/Documents/projects/tearribles/web/node_modules/.pnpm/@miniflare+core@2.13.0/node_modules/@miniflare/core/src/index.ts:790:13)
```

**Comment by frehner** [comment]
Hey everyone; for future reference, the Hydrogen team and the Remix team are different, and generally we aren't watching each other's repos for issues. 

In this case, we would much prefer that you open Hydrogen issues here instead: https://github.com/shopify/hydrogen 

---

To help solve your issue, though: 

When running Hydrogen locally (or when deployed on Oxygen), Hydrogen runs in a CloudFlare Workers-like environment. It seems Prisma has documentation on how to setup their client for "the Edge" (in other words, CloudFlare Workers / Oxygen / Hydrogen)

https://www.prisma.io/blog/database-access-on-the-edge-8F0t1s1BqOJE

The TLDR seems to be that you need to use the Prisma Data Proxy and `@prisma/client/edge` package

**Comment by Hector101** [comment]
@frehner Thanks for the assistance, but there seems to be another error being thrown after trying to setup with prisma proxy approach. Any idean what could be the cause?

```
ReferenceError: DATABASE_URL is not defined.
Attempted to access binding using global in modules.
You must use the 2nd `env` parameter passed to exported handlers/Durable Object constructors, or `context.env` with Pages Functions.
```

**Comment by machour** [comment]
Closing this issue, please continue this conversation on the hydrogen repo 🙏🏼

**Comment by MOuhax** [comment]
i am trying to build a shopify app and i have faced the same issue, Error coming from `npx prisma generate`.
